.github/workflows/docker.yaml

```diff
@@ -3,15 +3,92 @@ name: Docker
+        uses: actions/checkout@v2
+
+      - name: Set Git refname
+        id: set-git-refname
+        run: |
+          GIT_REFNAME="$(echo "${{ github.ref }}" | sed -r 's@refs/(heads|pull|tags)/@@g')"
+
+          echo "GIT_REFNAME=${GIT_REFNAME}"
+          echo ::set-output name=git_refname::${GIT_REFNAME}
+
+      - name: Set GitHub HEAD refname
+        id: set-github-head-refname
+        run: |
+          GITHUB_HEAD_REFNAME=""
+          case "${{ github.event_name }}" in
+            pull_request) GITHUB_HEAD_REFNAME="$(echo "${{ github.head_ref }}" | sed -r 's@refs/heads/@@g' )" ;;
+            *)            GITHUB_HEAD_REFNAME="${{ steps.set-git-refname.outputs.git_refname }}" ;;
+          esac
+
+          echo "GITHUB_HEAD_REFNAME=${GITHUB_HEAD_REFNAME}"
+          echo ::set-output name=github_head_refname::${GITHUB_HEAD_REFNAME}
+
+      - name: Set GitHub HEAD SHA
+        id: set-github-head-sha
+        run: |
+          GITHUB_HEAD_SHA=""
+          case "${{ github.event_name }}" in
+            pull_request) GITHUB_HEAD_SHA="$(git rev-parse "${{ github.head_ref }}")" ;;
+            *)            GITHUB_HEAD_SHA="${{ github.sha }}" ;;
+          esac
+
+          echo "GITHUB_HEAD_SHA=${GITHUB_HEAD_SHA}"
+          echo ::set-output name=github_head_sha::${GITHUB_HEAD_SHA}
+
+  # Note: I'm dissatisfied with the workflow_run event's current UX (depending workflow is not attached to commit/tag actions/checks).
+  wait-successful-ci-workflow:
+    name: Wait for CI workflow to finish successfully
+    needs:
+      - environment
+    runs-on: ${{ needs.environment.outputs.github_runner_default_os }} # Note: OS independent job.
+
+    steps:
+      - name: Wait for CI workflow to finish successfully
+        env:
+          CHECK_INTERVAL_SECONDS: 10
+          ENDPOINT: repos/${{ github.repository }}/actions/runs?branch=${{ needs.environment.outputs.github_head_refname }}&event=${{ github.event_name }}
+          GITHUB_WORKFLOW_CI_NAME: CI
+        run: |
+          echo "${{ github.token }}" | gh auth login --with-token
+
+          WORKFLOW_RUN="{}"
+          while echo "${WORKFLOW_RUN}" | jq --exit-status '(has("status") | not) or (.status != "completed")' &>/dev/null; do
+            echo "Waiting for the workflow to complete"
+            sleep ${CHECK_INTERVAL_SECONDS} # Note: waiting initially to ensure simultaneous workflow triggers have a chance to complete before accessing concurrent workflow.
+
+            WORKFLOW_RUN=$(gh api "${{ env.ENDPOINT }}" | jq '[ .workflow_runs[] | select((.name == "${{ env.GITHUB_WORKFLOW_CI_NAME }}") and (.head_branch == "${{ needs.environment.outputs.github_head_refname }}") and (.head_sha == "${{ needs.environment.outputs.github_head_sha }}")) ] | first')
+            echo "${WORKFLOW_RUN}" | jq
+          done
+
+          echo "${WORKFLOW_RUN}" | jq --exit-status 'has("status")  and (.status == "completed") and has("conclusion") and (.conclusion == "success")'
```

Comment:
I'm not sure what's the benefit of this. Ideally, we don't push images unless it's master or tags. Anything that hits master should already have a green CI workflow. If something breaks, it's still only master. Same for tags: we can always retract new tags if something goes wrong, but by then we should already have 2 green CI results.

Reply:
Fixed in 192d0b3 (removing job) and in caba4a7 (removing unneeded dependencies).